lib/internal/mime.js

```diff
@@ -182,103 +298,103 @@ class MIMEParams {
         // Unescape '\' quoted characters
-        parameterValue = removeBackslashes(inside);
-        // If we did have an unmatched '\' add it back to the end
-        if (insideMatch[1]) parameterValue += '\\';
+        parameterValue = collectHTTPQuotedString(str, marker);
+
+        // #11-8-1 Collect a sequence of code points that are not U+003B (;) from input, given position.
+        const remaining = StringPrototypeIndexOf(str, SEMICOLON, marker.position);
+        marker.position += remaining !== -1 ? remaining - marker.position : 0;
       } else {
-        // Handle the normal parameter value form
-        const valueEnd = StringPrototypeIndexOf(str, SEMICOLON, position);
+        // #11-9-1 Set parameterValue to the result of
+        // collecting a sequence of code points that are not U+003B (;) from input,
+        // given position.
+        const valueEnd = StringPrototypeIndexOf(str, SEMICOLON, marker.position);
         const rawValue = valueEnd === -1 ?
-          StringPrototypeSlice(str, position) :
-          StringPrototypeSlice(str, position, valueEnd);
-        position += rawValue.length;
-        const trimmedValue = StringPrototypeSlice(
-          rawValue,
-          0,
-          SafeStringPrototypeSearch(rawValue, START_ENDING_WHITESPACE),
-        );
-        // Ignore parameters without values
-        if (trimmedValue === '') continue;
+          StringPrototypeSlice(str, marker.position) :
+          StringPrototypeSlice(str, marker.position, valueEnd);
+
+        // #11-9-2 Remove any trailing HTTP whitespace from parameterValue.
+        const trimmedValue = StringPrototypeTrimEnd(rawValue);
+        marker.position += trimmedValue.length;
+
+        // #11-9-3 If parameterValue is the empty string, then continue.
+        if (trimmedValue.length === 0) continue;
+
         parameterValue = trimmedValue;
       }
+
+      /**
+       * #11-10 if:
+       * parameterName is not the empty string
+       * parameterName solely contains HTTP token code points
+       * parameterValue solely contains HTTP quoted-string token code points
+       * mimeType parameters[parameterName] does not exist
+       * then set mimeType parameters[parameterName] to parameterValue.
+       */
       if (
-        parameterString !== '' &&
-        SafeStringPrototypeSearch(parameterString,
-                                  NOT_HTTP_TOKEN_CODE_POINT) === -1 &&
-        SafeStringPrototypeSearch(parameterValue,
-                                  NOT_HTTP_QUOTED_STRING_CODE_POINT) === -1 &&
-        params.has(parameterString) === false
+        parameterName.length !== 0 &&
+        SafeStringPrototypeSearch(
+          parameterName,
+          NOT_HTTP_TOKEN_CODE_POINT,
+        ) === -1 &&
+        SafeStringPrototypeSearch(
+          parameterValue,
+          NOT_HTTP_QUOTED_STRING_CODE_POINT,
+        ) === -1 &&
+        params.has(parameterName) === false
```

Comment:
isnt .has on Map O(1)?

```suggestion
        params.has(parameterName) === false &&
        SafeStringPrototypeSearch(
          parameterValue,
          NOT_HTTP_QUOTED_STRING_CODE_POINT,
        ) === -1
```

Reply:
Yeah, but I'm afraid I'm not following 🤔